src/pages/commits/commitChart/CommitChart.tsx

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import styled from "@emotion/styled";
+import { uiColors } from "@leafygreen-ui/palette";
+
+const { green, gray, yellow, red } = uiColors;
+
+export type TaskStats = {
+  success?: number;
+  failure?: number;
+  dispatched?: number;
+  scheduled?: number;
+  unscheduled?: number;
+  systemFailure?: number;
+  setupFailure?: number;
+  total: number;
+};
+
+interface Props {
+  taskStats: TaskStats;
+  max: number;
+  chartType: "percentage" | "absolute";
+}
+
+function calculateHeight(
+  value: number,
+  max: number,
+  total: number,
+  chartType: string
+) {
+  if (chartType === "percentage") {
+    return `${(value / total) * 100}%`;
+  }
+  return `${(value / max) * 100}%`;
+}
+export const CommitChart: React.FC<Props> = ({ taskStats, max, chartType }) => {
+  const { total } = taskStats || {};
+  return (
+    <ChartContainer>
+      {Object.keys(taskStats).map((status) => (
+        <Bar
+          height={calculateHeight(taskStats[status], max, total, chartType)}
+          color={barColors[status]}
+        />
+      ))}
+    </ChartContainer>
+  );
+};
+
+const barColors = {
+  success: green.base,
+  failure: red.base,
+  dispatched: yellow.base,
+  scheduled: gray.dark1,
+  unscheduled: gray.light1,
+  systemFailure: "#633F70",
+  setupFailure: "#A075AF",
```

Comment:
Can you use this enum for the TaskStatuses
https://github.com/evergreen-ci/spruce/blob/d2e0865ab48c5f8f83462a98ec12b20eaaa59f96/src/types/task.ts#L51

We also have this constants file with color mappings can we use those?
https://github.com/evergreen-ci/spruce/blob/d2e0865ab48c5f8f83462a98ec12b20eaaa59f96/src/constants/task.ts#L137

Reply:
Although the colors are slightly different we plan on consolidating them soon so they will eventually be the same.